src/components/DistanceRequest/index.js

```diff
@@ -0,0 +1,235 @@
+            return;
+        }
+
+        // Create the initial start and stop waypoints
+        Transaction.createInitialWaypoints(transactionID);
+    }, [transaction, transactionID]);
+
+    useEffect(() => {
+        if (isOffline || !shouldFetchRoute) {
+            return;
+        }
+
+        Transaction.getRoute(transactionID, validatedWaypoints);
+    }, [shouldFetchRoute, transactionID, validatedWaypoints, isOffline]);
+
+    useEffect(() => {
+        if (numberOfWaypoints <= numberOfPreviousWaypoints) {
+            return;
+        }
+        scrollViewRef.current.scrollToEnd({animated: true});
+    }, [numberOfPreviousWaypoints, numberOfWaypoints]);
+
+    const navigateBack = () => {
+        Navigation.goBack(isEditing ? ROUTES.MONEY_REQUEST_CONFIRMATION.getRoute(iouType, reportID) : ROUTES.HOME);
+    };
+
+    /**
+     * Takes the user to the page for editing a specific waypoint
+     * @param {Number} index of the waypoint to edit
+     */
+    const navigateToWaypointEditPage = (index) => {
+        Navigation.navigate(isEditingRequest ? ROUTES.MONEY_REQUEST_EDIT_WAYPOINT.getRoute(report.reportID, transactionID, index) : ROUTES.MONEY_REQUEST_WAYPOINT.getRoute('request', index));
+    };
+
+    const updateWaypoints = useCallback(
+        ({data}) => {
+            if (_.isEqual(waypointsList, data)) {
+                return;
+            }
+
+            const newWaypoints = {};
+            _.each(data, (waypoint, index) => {
+                const newWaypoint = lodashGet(waypoints, waypoint, {});
+                newWaypoints[`waypoint${index}`] = lodashIsEmpty(newWaypoint) ? null : newWaypoint;
+            });
+
+            setOptimisticWaypoints(newWaypoints);
+            // eslint-disable-next-line rulesdir/no-thenable-actions-in-views
+            Transaction.updateWaypoints(transactionID, newWaypoints).then(() => {
+                setOptimisticWaypoints(null);
+            });
+        },
+        [transactionID, waypoints, waypointsList],
+    );
+
+    const content = (
+        <>
+            <View style={styles.flex1}>
+                <DraggableList
+                    data={waypointsList}
+                    keyExtractor={(item) => item}
+                    shouldUsePortal
+                    onDragEnd={updateWaypoints}
+                    scrollEventThrottle={variables.distanceScrollEventThrottle}
+                    ref={scrollViewRef}
+                    renderItem={({item, drag, isActive, getIndex}) => (
+                        <DistanceRequestRenderItem
+                            waypoints={waypoints}
+                            item={item}
+                            onSecondaryInteraction={drag}
+                            isActive={isActive}
+                            getIndex={getIndex}
+                            onPress={navigateToWaypointEditPage}
+                            disabled={isLoadingRoute}
```

Comment:
@blazejkustra Sorry. I have a minor question. Please help answer when you have a chance. Why do we need this logic?

Reply:
@DylanDylann This line disables the items of draggable list when the route is reloading, so that it is impossible to reorder the list while the map is reloading. Without it, there is a possible race condition if you change the order of items quickly while the map is still loading the route.

Reply:
Many thanks @blazejkustra